x/mint/types/minter.go

```diff
@@ -49,42 +49,50 @@ func DefaultInitialMinter() Minter {
 	return InitialMinter()
 }
 
-// ValidateMinter validate minter.
+// ValidateMinter ensure minter has valid "normTimePassed" and
+// "totalMinted" tokens do not exceed the minting cap.
 func ValidateMinter(minter Minter) error {
 	if minter.NormTimePassed.IsNegative() {
 		return fmt.Errorf("mint parameter normTimePassed should be positive, is %s",
 			minter.NormTimePassed.String())
 	}
+
 	if minter.NormTimePassed.GT(TotalMonths) {
 		return fmt.Errorf("mint parameter normTimePassed: %v should not be bigger than TotalMonths: %v", minter.NormTimePassed, TotalMonths)
 	}
+
 	if minter.TotalMinted.GT(MintingCap) {
-		return fmt.Errorf("mint parameter totalMinted can not be bigger than MintingCap, is %s",
-			minter.TotalMinted)
+		return fmt.Errorf("mint parameter totalMinted: %v can not be bigger than MintingCap: %v",
+			minter.TotalMinted, MintingCap)
 	}
+
 	calculatedMintedTokens := calcMintedTokens(minter)
+
 	if minter.NormTimePassed.GT(TotalMonths.Sub(sdk.NewDec(1))) {
 		if calculatedMintedTokens.GT(MintingCap) || MintingCap.Sub(calculatedMintedTokens).GT(FixedMintedAmount) {
 			return fmt.Errorf("mint parameters are not conformant with the minting schedule, for %s month minted %s unls",
 				minter.NormTimePassed, calculatedMintedTokens)
 		}
 	} else if !calculatedMintedTokens.Equal(minter.TotalMinted) {
-		return fmt.Errorf("minted unexpected amount of tokens for %s months: %s unls",
-			minter.NormTimePassed, minter.TotalMinted)
+		return fmt.Errorf("minted unexpected amount of tokens for %s months. act: %v, exp: %v",
+			minter.NormTimePassed, minter.TotalMinted, calculatedMintedTokens)
 	}
 
 	return nil
 }
 
 func calcMintedTokens(m Minter) sdk.Uint {
-	fixedMonthsTokens := sdk.NewUint(0)
-	calculatedTokensByIntegral := sdk.NewUint(0)
+	fixedMonthsTokens := sdk.ZeroUint()
```

Comment:
this line is better to be moved into the `if` clause to reduce its scope and make the code more clear